spec/lib/workflow/stores/mongo-spec.js

```diff
@@ -239,6 +239,36 @@ describe('Task Graph mongo store interface', function () {
         });
     });
 
+    it('persistTaskDefinition should recursively replace keywords if any', function() {
+        var definition = {
+            injectableName: 'testname',
+            optionsSchema: {
+                version: {
+                    $ref: "abc.json"
```

Comment:
In unit-test, please add some more complex example to test the recursive works well